Added method to look for new api in fakes datacollector

vritant24 opened the pull request:
## Description
The new Fakes Implementation will need a separate data collector for handling new scenarios that support .NET Core and Netstandard.
Previously, `TestWindow.Fakes` and `VSTest` were making the same check and passing the same data collector configuration (`FriendlyName`, `DataCollectorURi`, `DataCollectorAssemblyQualifiedName`) to `TestPlatform.Fakes`. The creation of that config will be moved into `TestPlatform.Fakes`, and the decision to use the new data collector will also be made there, which introduces the new API.

vritant24 committed:
Added method to look for new api in fakes datacollector

vritant24 committed:
Added method to look for new api in fakes datacollector

vritant24 committed:
added check for .net runtime version

vritant24 committed:
added check for .net runtime version

vritant24 committed:
refactored code

vritant24 committed:
refactored code

vritant24 committed:
enable new configurator for net core

vritant24 committed:
enable new configurator for net core

vritant24 committed:
refactored to remove use of out param

vritant24 committed:
refactored to remove use of out param

vritant24 commented on src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs:
```diff
@@ -168,6 +191,34 @@ private static bool TryGetFakesDataCollectorConfigurator(out Func<IEnumerable<st
+                Assembly assembly = Assembly.Load(FakesConfiguratorAssembly);
+
+                var type = assembly?.GetType(ConfiguratorAssemblyQualifiedName, false);
+                if (type != null)
```
Resolved

vritant24 commented on src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs:
```diff
@@ -168,6 +191,34 @@ private static bool TryGetFakesDataCollectorConfigurator(out Func<IEnumerable<st
+                    var method = type.GetMethod(ConfiguratorMethodName, BindingFlags.Public | BindingFlags.Static);
+                    if (method != null)
+                    {
+                        configurator = (Func<IEnumerable<string>, bool, DataCollectorSettings>)method.CreateDelegate(typeof(Func<IEnumerable<string>, bool, DataCollectorSettings>));
```
I went ahead and refactored to remove the `out` params.

nohwnd commented on src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs:
```diff
@@ -77,31 +71,59 @@ private static bool IsNetCoreFramework(string runSettingsXml)
             }
 
-            Func<IEnumerable<string>, string> configurator;
+            // A new Fakes Congigurator API makes the decision to add the right datacollector uri to the configuration
```
nit: scanrio  -> scenario

nohwnd commented on src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs:
```diff
@@ -77,31 +71,59 @@ private static bool IsNetCoreFramework(string runSettingsXml)
 
             // integrate fakes settings in configuration
-            // if the settings don't have any data collector settings, populate with empty settings
+            // if the settings doesn't have any data collector settings, populate with empty settings
```
I think you were right when you used don't

AbhitejJohn replied:
This has been updated hence too

nohwnd commented on src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs:
```diff
@@ -164,8 +185,33 @@ private static bool TryGetFakesDataCollectorConfigurator(out Func<IEnumerable<st
+            try
+            {
+                Assembly assembly = Assembly.Load(FakesConfiguratorAssembly);
+
```
nit: are the extra white lines necessary?

AbhitejJohn replied:
Looks like that's fixed.

nohwnd commented:
@vritant24 is this ready to merge? the changes look good to go and I approved. If you will still work on it please mark it as work in progress by prefixing the name with `WIP: ` it makes it easier for me to see that I should not re-visit this when looking at PRs. 🙂

vritant24 commented:
Thanks for the review. I haven't yet merged it because the API on the other side hasn't been finalized and could possibly change. I'll update the PR name

AbhitejJohn committed:
Removed Microsoft.VisualStudio.TestPlatform.Fakes from the "Microsoft.Internal.TestPlatform.Extensions" nuget pacakge. This would mean that the TP vsix would not drop this dependency any more. The Fakes vsix that builds in the VSUT repo would be dropping it instead.
For the TestPlatform tooling nuget however which ships with the Fakes dependency, I'm pulling in the latest from the vsunittesting myget feed and packaging that up.

AbhitejJohn committed:
Removed Microsoft.VisualStudio.TestPlatform.Fakes from the "Microsoft.Internal.TestPlatform.Extensions" nuget pacakge. This would mean that the TP vsix would not drop this dependency any more. The Fakes vsix that builds in the VSUT repo would be dropping it instead.
For the TestPlatform tooling nuget however which ships with the Fakes dependency, I'm pulling in the latest from the vsunittesting myget feed and packaging that up.

AbhitejJohn committed:
Fixing method resolution.

AbhitejJohn committed:
Fixing method resolution.

nohwnd commented on scripts/build.ps1:
```diff
@@ -570,7 +570,7 @@ function Create-VsixPackage
 
     # Copy legacy dependencies
-    $legacyDir = Join-Path $env:TP_PACKAGES_DIR "Microsoft.Internal.TestPlatform.Extensions\$testPlatformExternalsVersion-patched3\contentFiles\any\any"
+    $legacyDir = Join-Path $env:TP_PACKAGES_DIR "Microsoft.Internal.TestPlatform.Extensions\$testPlatformExternalsVersion-patched5\contentFiles\any\any"
```
@AbhitejJohn  We talked about this on call, just a reminder so this does not go unfixed. 🙂  Please base this package on patched4 package, because that includes the correct symbols for trace data collector.

AbhitejJohn replied:
yup, already did so. I've deleted the old patched-5 to avoid confusion and pushed up an updated version based on patched-4.

vritant24 committed:
updated api call

vritant24 committed:
updated api call

vritant24 committed:
fixed pr comments

vritant24 committed:
fixed pr comments

vritant24 committed:
merged conflict

AbhitejJohn committed:
Fixing build - this doesn't need to be signed.

AbhitejJohn committed:
Fixing build - this doesn't need to be signed.

vritant24 committed:
removed obsolete

vritant24 committed:
removed obsolete

AbhitejJohn commented on src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs:
```diff
@@ -164,8 +186,30 @@ private static bool TryGetFakesDataCollectorConfigurator(out Func<IEnumerable<st
+            {
+                if (EqtTrace.IsInfoEnabled)
+                {
+                    EqtTrace.Info("Failed to create newly implemented Fakes Configurator. Reason:{0} ", ex);
```
Does this show up in test platform diagnostic logs? Its probably used in some places in this file already but eqttrace logs IIRC are enabled and viewed differently from how we enable TP diagnostic logs in general.

AbhitejJohn committed:
Merge branch 'master' into dev/vrbhardw/fakes_dc_update

AbhitejJohn committed:
Looks like the signed build has a different way of producing nuget packages from the dev build.

vritant24 committed:
removed system.linq reference:

vritant24 committed:
fixed rebase errors

vritant24 committed:
merged remote

AbhitejJohn committed:
Fixing script vars

AbhitejJohn committed:
Merge branch 'dev/vrbhardw/fakes_dc_update' of https://github.com/microsoft/vstest into dev/vrbhardw/fakes_dc_update

nohwnd committed:
Write value to the correct pipeline variable
